src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -517,8 +515,8 @@ def left_semi_or_anti_join_statement(
 def snowflake_supported_join_statement(
     left: str, right: str, join_type: JoinType, condition: str
 ) -> str:
-    left_alias = random_name_for_temp_object(TempObjectType.TABLE)
-    right_alias = random_name_for_temp_object(TempObjectType.TABLE)
+    left_alias = "SNOWPARK_TEMP_LEFT_RESULT"
```

Comment:
Some queries may have many repeated query aliases if we use a fixed one. For instance, in the following code df7's query will have two SNOWPARK_TEMP_LEFT_RESULT as subquery aliases. How about removing the alias if we use this way.
```
df5 = df1 .join(df2)
df6 = df3.join(df4)
df7 = df5.join(df6)
```

Reply:
That's a great point, why do we need these subquery aliases? I don't see them being used or saved separately from the query string 🤔

Reply:
I don't have a strong opinion here but I think a longer name with `SNOWPARK` in it is likely to be duplicated.